matrix/utils.py

```diff
@@ -31,16 +44,18 @@ def __call__(cls, *args, **kwargs):
         return cls._instances[cls]
 
 
-def deepmerge(dest, src):
+def deepmerge(dest, src, merge_lists=False):
     """
     Deep merge of two dicts.
 
     This is destructive (`dest` is modified), but values
     from `src` are passed through `copy.deepcopy`.
     """
     for k, v in src.items():
-        if dest.get(k) and isinstance(v, dict):
-            deepmerge(dest[k], v)
+        if isinstance(v, dict):
+            deepmerge(dest.setdefault(k, {}), v)
+        elif isinstance(v, list) and merge_lists:
```

Comment:
Merging lists is almost never a good idea, order and uniqueness are key reasons for using lists, here we'd have to give up one or the other.